modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -769,7 +769,7 @@ private void generateSupportingFiles(List<File> files, Map<String, Object> bundl
         bundle.put("basePathWithoutHost", basePathWithoutHost);
         bundle.put("scheme", URLPathUtils.getScheme(url, config));
         bundle.put("host", url.getHost());
-        if (url.getPort() != 80 && url.getPort() != 443 ) {
+        if (url.getPort() != 80 && url.getPort() != 443 && url.getPort() != -1) {
```

Comment:
I guess that does the trick but why was the port `-1` to begin with? Some default value? Unfortunate conversion or something?
If there's no way to make it so it doesn't get the `-1` value I suppose a comment explaining why `-1` is a possible value and needs to be handled would help in the future. 
Port 80 and 443 people can figure it out, -1, not so much I guess.

Reply:
My guess is that it's a "special logic/convention" in swagger parser but don't have time to dig into that code base to confirm.

Reply:
> The getPort() function is a part of URL class. The function getPort() returns the port of a specified URL. The function returns the port number or -1 if the port is not set

Ref: https://www.geeksforgeeks.org/url-getport-method-in-java-with-examples/